Error in headless Chromium

When running the headless tests in Chromium there is an error.

```
npm run test-headless

> @sinonjs/samsam@6.1.1 test-headless
> mochify --no-detect-globals --recursive -R dot --plugin [ proxyquire-universal ] "./lib/*.test.js"

# chromium:


  ....................................................................................................................................................................................................................
  .....................................................................................................................................................................................................!..............
  .....................................................................................................................................,...................................

  591 passing (64ms)
  1 pending
  1 failing

  1) issues
       issue #71
         when comparing different jQuery objects
           should return false:
     AssertionError: [assert.isFalse] Expected true to be false
      at Object.fail (node_modules/@sinonjs/referee/lib/create-fail.js:5)
      at Object.fail (node_modules/@sinonjs/referee/lib/define-assertion/index.js:47)
      at assertion (node_modules/@sinonjs/referee/lib/define-assertion/index.js:65)
      at Function.referee.<computed>.<computed> [as isFalse] (node_modules/@sinonjs/referee/lib/define-assertion/index.js:92)
      at Context.<anonymous> (lib/issues.test.js:27)



Error: Exit 1
```

--- a/lib/deep-equal.js
+++ b/lib/deep-equal.js
@@ -191,6 +191,19 @@ function deepEqualCyclic(actual, expectation, match) {
             return mapsDeeplyEqual;
         }
 
+        // jQuery objects have iteration protocols
+        // see: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Iteration_protocols
+        // But, they don't work well with the implementation concerning iterables below,
+        // so we will detect them and use jQuery's own equality function
+        /* istanbul ignore next -- this can only be tested in the `test-headless` script */
+        if (
+            actualObj.constructor &&
+            actualObj.constructor.name === "jQuery" &&
+            typeof actualObj.is === "function"
+        ) {
+            return actualObj.is(expectationObj);
+        }
+
         var isActualNonArrayIterable =
             isIterable(actualObj) &&
             !isArrayType(actualObj) &&

--- a/lib/issues.test.js
+++ b/lib/issues.test.js
@@ -24,8 +24,15 @@ describe("issues", function () {
 
         context("when comparing different jQuery objects", function () {
             it("should return false", function () {
+                jQuery(`
+                    <body>
+                        <p id='test1'>My <em>old</em> text</p>
+                        <p id='test2'>My <em>new</em> text</p>
+                    </body>
+                `).appendTo("html");
+
                 assert.isFalse(
-                    samsam.deepEqual(jQuery("head"), jQuery("body"))
+                    samsam.deepEqual(jQuery("#test1"), jQuery("#test2"))
                 );
             });
         });

